### PR TITLE
Frame-work for the LSST-event-broker

### PR DESCRIPTION
**Pull request by csumano**
The documentation may not be accurate or even right, but please feel free to add or change anything if it is not correct.

### CODE REVIEW

File: Tests/test_main.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# import
```
Comment (ExcaliburZero) [inline Tests/test_main.py]: We don't need this file yet, so you can remove it for now. We will need something like this once we start to add functionality to the skeleton.

File: lsstbroker/BinaryClassifier.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"""
```
Comment (ExcaliburZero) [inline lsstbroker/BinaryClassifier.py]: You can remove this empty docstring.

File: lsstbroker/BinaryClassifier.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+
+
+"""
+
+
+class BinaryClassifer():
+
```
Comment (ExcaliburZero) [inline lsstbroker/BinaryClassifier.py]: Add a variable classifying_function here.

``` python
classifying_function = None
```

File: lsstbroker/classifier.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+"""
+
+
+"""
+
+
+class Classifier():
+
```
Comment (ExcaliburZero) [inline lsstbroker/classifier.py]: Add these

``` python
determining_classifier = None
binary_classifiers = []
```

File: lsstbroker/classifier.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+"""
```
Comment (ExcaliburZero) [inline lsstbroker/classifier.py]: Remove this.

File: lsstbroker/classifierBox.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+"""
```
Comment (ExcaliburZero) [inline lsstbroker/classifierBox.py]: Remove this.

File: lsstbroker/classifierBox.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+"""
+
+
+
+class ClassifierBox():
+
```
Comment (ExcaliburZero) [inline lsstbroker/classifierBox.py]: Add this:

``` python
classifiers = []
```

File: lsstbroker/transientObservation.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+This class will get the time value and color observation from each transient
+and store that data into each specific transient. Afterwards this data can be used
+to classify the data based on this information.
+
+"""
+
+
+class transientOberservation(object):
+
```
Comment (ExcaliburZero) [inline lsstbroker/transientObservation.py]: Add this:

``` python
time = None
light = None
color = None
```

**Review by ExcaliburZero — CHANGES_REQUESTED** [review]
Looks good, just needs a few small changes.

File: lsstbroker/BinaryClassifier.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
+class BinaryClassifer():
```
Comment (ExcaliburZero) [inline lsstbroker/BinaryClassifier.py]: Rename this file to `binary_classifier.py`

File: lsstbroker/classifierBox.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
+class ClassifierBox():
```
Comment (ExcaliburZero) [inline lsstbroker/classifierBox.py]: Rename this file to `classifier_box.py`

File: lsstbroker/transientObservation.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
+"""
```
Comment (ExcaliburZero) [inline lsstbroker/transientObservation.py]: Rename this file to `transient_observation.py`.

File: lsstbroker/BinaryClassifier.py (hunk 9 above)
Comment (ExcaliburZero) [inline lsstbroker/BinaryClassifier.py]: Change this line to: (New style class)

``` python
class BinaryClassifer(object):
```

File: lsstbroker/classifier.py [hunk 12]
```diff
@@ -0,0 +1,27 @@
+class Classifier():
```
Comment (ExcaliburZero) [inline lsstbroker/classifier.py]: Change this line to:

``` python
class Classifier(object):
```

File: lsstbroker/classifier.py [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+
+    determining_classifier = None
+    binary_classifiers = []
+
+    def setDeterminingClassifier(binary_classifier):
```
Comment (ExcaliburZero) [inline lsstbroker/classifier.py]: Change this to: (Python naming convention + Python methods take in self / this)

``` python
def set_determining_classifier(self, binary_classifier):
```

File: lsstbroker/classifier.py [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        """
+        This method determines the classifier
+        """
+
+    def addBinaryClassifier(binary_classifier):
```
Comment (ExcaliburZero) [inline lsstbroker/classifier.py]: Change this to:

``` python
def add_binary_classifier(self, binary_classifier):
```

**Review by csumano — COMMENTED** [review]
i cleaned it up and made the changes

File: lsstbroker/classifier.py [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+        """
+        This method adds any diffirent binary classifiers to the
+        given object
+        """
+
+    def run(observations):
```
Comment (ExcaliburZero) [inline lsstbroker/classifier.py]: Change this to:

``` python
def run(self, observations):
```

File: lsstbroker/classifierBox.py (hunk 10 above)
Comment (ExcaliburZero) [inline lsstbroker/classifierBox.py]: Change this to:

``` python
class ClassifierBox(object):
```

File: lsstbroker/classifierBox.py [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+    classifiers = []
+
+    def addclassifier(classifier):
```
Comment (ExcaliburZero) [inline lsstbroker/classifierBox.py]: Change this to:

``` python
def add_classifier(self, classifier):
```

File: lsstbroker/classifierBox.py [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+        """
+        This method adds a specific classifier to the ClassifierBox
+        """
+
+    def run(observations):
```
Comment (ExcaliburZero) [inline lsstbroker/classifierBox.py]: Change this to:

``` python
def run(self, observations):
```

File: lsstbroker/transientObservation.py [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+
+This class will get the time value and color observation from each transient
+and store that data into each specific transient. Afterwards this data can be used
+to classify the data based on this information.
+"""
+
+class transientOberservation(object):
```
Comment (ExcaliburZero) [inline lsstbroker/transientObservation.py]: Change this to:

``` python
class TransientOberservation(object):
```

**Review by ExcaliburZero — CHANGES_REQUESTED** [review]
Okay, just a few style changes. I messed up some of the naming in the schema. Python naming is a bit different.

File: lsstbroker/BinaryClassifier.py [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+
+    classifying_function = None
+
+    def __init__(classifying_function):
```
Comment (ExcaliburZero) [inline lsstbroker/BinaryClassifier.py]: Change this to:

``` python
def __init__(self, classifying_function):
```

File: lsstbroker/transientObservation.py [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+
+    time = None
+    light = None
+    color = None
+
+    def __init__(time, light_value, color) : # init the time light value and color
```
Comment (ExcaliburZero) [inline lsstbroker/transientObservation.py]: Change this to:

``` python
def __init__(self, time, light_value, color):
```

File: lsstbroker/binary_classifier.py [hunk 21]
```diff
@@ -0,0 +1,14 @@
+class BinaryClassifer(object):
+
+    classifying_function = None
+
+    def __init__(self, classifying_function):
+        """
+        Constructor for the classification
+        """
+
+    def run(observations):
```
Comment (ExcaliburZero) [inline lsstbroker/binary_classifier.py]: Change this to:

``` python
def run(self, observations):
```

File: lsstbroker/classifier_box.py [hunk 22]
```diff
@@ -0,0 +1,14 @@
+class ClassifierBox(object):
+
+    classifiers = []
+
+    def addclassifier(self, classifier):
```
Comment (ExcaliburZero) [inline lsstbroker/classifier_box.py]: Change this to:

``` python
def add_classifier(self, classifier):
```

File: lsstbroker/transient_observation.py [hunk 23]
```diff
@@ -0,0 +1,20 @@
+"""
+
+This class will get the time value and color observation from each transient
+and store that data into each specific transient. Afterwards this data can be used
+to classify the data based on this information.
+"""
+
+class transientOberservation(object):
```
Comment (ExcaliburZero) [inline lsstbroker/transient_observation.py]: Change this to:

``` python
class TransientOberservation(object):
```

**Review by ExcaliburZero — APPROVED** [review]

**Comment by ExcaliburZero** [comment]
Merged in eb1665c2b3bd8df441fc42c9fe5399f615e64993

**Pull request closed without merging**